ArduPlane/parachute.cpp

```diff
@@ -56,3 +59,55 @@ bool Plane::parachute_manual_release()
 
     return true;
 }
+
+// Code to detect loss of control for ArduPlane
+#define PARACHUTE_CHECK_TRIGGER_SEC         1  // 1 second of loss of control triggers the parachute
+
+/*
+  parachute_emergency_check - trigger the release of the parachute
+  automatically if a critical situation is detected
+*/
+void Plane::parachute_emergency_check()
+{
+    uint32_t now = millis();
+
+    // exit immediately if parachute or automatic release is not enabled, or already released
+    if (!parachute.enabled() || parachute.released() || g.parachute_auto_enabled == 0) {
+        chute_control_loss_ms = 0;
```

Comment:
move chute_control_loss_ms to be an AP_Parachute member variable with a set/get accessor so this line would be parachute.set_chute_control_loss_ms(0). The goal is to push as much as possible into the library. This IF should also get turned into a single parachute.is_auto_enabled() that makes these checks in the lib.